### PR TITLE
 Make a 1.0 only branch/release

### PR DESCRIPTION
**Pull request by oxinabox**
Things brings support for 1.0, and targets the `release-1.0` branch.

It is actually much simpler than for later versions.
Since no test/Project.toml to worry about.
And `with_dependencies_loadable_at_toplevel` has semantics that let us just copy the the files we need out of the sandbox without activating it.
I am yet to see how to apply that trick in modern julia versions, but it might be possible.

Anyway, this PR deletes a bunch of `if VERSION checks`, and refactors the code, similar to #8

### CODE REVIEW

File: test/activate_do.jl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@testset "activate_do.jl" begin
+    @testset "activate do f [extras]" begin
+        TestEnv.activate("ChainRulesCore") do
+            @eval using FiniteDifferences
+        end
+        @test isdefined(@__MODULE__, :FiniteDifferences)
+    end
+end
```
Comment (mmiller-max) [inline test/activate_do.jl:8]: Do you want to check the original project is unchanged here?

**Review by mmiller-max — APPROVED** [review]

File: src/common.jl [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            pkgfilepath = find_installed(pkgspec.name, pkgspec.uuid, SHA1(info["git-tree-sha1"]))
+        elseif haskey(info, "path")
+            pkgfilepath =  project_rel_path(ctx, info["path"])
+        elseif pkgspec.uuid in keys(ctx.stdlibs)
+            pkgfilepath = Pkg.Types.stdlib_path(pkgspec.name)
+        else
+            throw(TestEnvError("Could not find either `git-tree-sha1` or `path` for package $(pkgspec.name)"))
+        end
+    end
+    return joinpath(pkgfilepath, "test")
```
Comment (mmiller-max) [inline src/common.jl:73]: Could delete this line as per `v1.1` branch

**Review by mmiller-max — APPROVED** [review]

File: src/common.jl (hunk 2 above)
Comment (oxinabox) [reply src/common.jl:73]: If we don't have it the function doesn't do what it says it does.
Not that we use this function's actual effect in v1.0-1.2, we only use it's side-effect

File: test/activate_do.jl [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@testset "activate_do.jl" begin
+    @testset "activate do f [extras]" begin
+        TestEnv.activate("ChainRulesCore") do
+            @eval using FiniteDifferences
+        end
+        @test isdefined(@__MODULE__, :FiniteDifferences)
```
Comment (oxinabox) [inline test/activate_do.jl:6]:
```suggestion
        orig_project = Base.active_project()
        TestEnv.activate("ChainRulesCore") do
            @eval using FiniteDifferences
        end
        @test isdefined(@__MODULE__, :FiniteDifferences)
        @test Base.active_project() == orig_project
```

**Pull request merged**